src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProjectTracker.cs

```diff
@@ -282,7 +289,7 @@ internal void AddProject(AbstractProject project)
             {
                 StartPushingToWorkspaceAndNotifyOfOpenDocuments(SpecializedCollections.SingletonEnumerable(project));
             }
-            else
+            else if (_batchingProjectLoads)
```

Comment:
I see, it's already in `_projectMap`, and in `FinishLoad` we push everything thats in `_projectMap`.